sdk/nginx/Azure.ResourceManager.Nginx/tests/NginxManagementTestBase.cs

```diff
@@ -0,0 +1,315 @@
+            return lro.Value;
+        }
+
+        protected string GetPublicIPAddress(PublicIPAddressResource publicIP)
+        {
+            if (publicIP == null)
+            {
+                throw new ArgumentNullException(nameof(publicIP));
+            }
+
+            return publicIP.Data.IPAddress;
+        }
+
+        protected async Task<NginxDeploymentResource> CreateNginxDeployment(ResourceGroupResource resourceGroup, AzureLocation location, string nginxDeploymentName)
+        {
+            if (resourceGroup == null)
+            {
+                throw new ArgumentNullException(nameof(resourceGroup));
+            }
+
+            if (nginxDeploymentName == null)
+            {
+                throw new ArgumentNullException(nameof(nginxDeploymentName));
+            }
+
+            PublicIPAddressResource publicIP = await CreatePublicIP(resourceGroup, location);
+            NginxFrontendIPConfiguration frontEndIPConfiguration = new NginxFrontendIPConfiguration();
+            frontEndIPConfiguration.PublicIPAddresses.Add(new WritableSubResource
+            {
+                Id = publicIP.Data.Id
+            });
+
+            NetworkSecurityGroupResource nsg = await CreateNetworkSecurityGroup(resourceGroup, location);
+            VirtualNetworkResource vnet = await CreateVirtualNetwork(resourceGroup, location, nsg.Data);
+            string subnetId = GetSubnetId(vnet);
+
+            NginxNetworkProfile networkProfile = new NginxNetworkProfile();
+            networkProfile.FrontEndIPConfiguration = frontEndIPConfiguration;
+            networkProfile.NetworkInterfaceSubnetId = subnetId;
+
+            NginxDeploymentProperties deploymentProperties = new NginxDeploymentProperties();
+            deploymentProperties.NetworkProfile = networkProfile;
+            deploymentProperties.EnableDiagnosticsSupport = true;
+
+            ManagedServiceIdentity identity = new ManagedServiceIdentity(ManagedServiceIdentityType.UserAssigned);
+            identity.UserAssignedIdentities.Add(new ResourceIdentifier(TestEnvironment.ManagedIdentityResourceID), new UserAssignedIdentity());
+
+            NginxDeploymentData nginxDeploymentData = new NginxDeploymentData(location);
+            nginxDeploymentData.Identity = identity;
+            nginxDeploymentData.Properties = deploymentProperties;
+            nginxDeploymentData.SkuName = "preview_Monthly";
+            ArmOperation<NginxDeploymentResource> lro = await resourceGroup.GetNginxDeployments().CreateOrUpdateAsync(WaitUntil.Completed, nginxDeploymentName, nginxDeploymentData);
+            return lro.Value;
+        }
+
+        protected async Task<NginxConfigurationResource> CreateNginxConfiguration(ResourceGroupResource resourceGroup, AzureLocation location, NginxDeploymentResource nginxDeployment, string nginxConfigurationName, string virtualPath)
```

Comment:
I didn't see the parameter `resourceGroup` being used in this method, maybe we can delete it?

Reply:
Done